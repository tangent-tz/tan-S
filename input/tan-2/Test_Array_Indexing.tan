main {
    var x1 := "start";  
    print x1 \n; 
    
    const arrayLen := 7;
    
    # TEST INT ARRAY =========================================================
    var x2 := [1, 2, 33, 4, 5, 6, 777777]; 
    print x2 \n; 
    
    var i := 0;
    while (i < length x2) {
        print [x2:i] \t; 
        i := i + 1; 
    }
    print \n; 
    
    print [x2:length x2-1] != 5 \n; 
    print [x2:length x2-1] <= 1000000 \n; 
    print [x2:length x2-1] >= 1000000 \n;
    print \n; 
    
    
    
    # TEST FLOAT ARRAY =========================================================
    var x3 := [1.34, 1.0 , 33.2, 4.4, 5.567, -6.5456, -0.20]; 
    print x3 \n; 
    
    i := 0;
    while (i < length x3) {
        print [x3:i] \t; 
        i := i + 1; 
    }
    print \n;  
    
    print [x3:1] == 0.5 + 0.5 \n; 
    print [x3:1] > -1 \n; 
    print \n; 
    
    
    
    # TEST STRING ARRAY =========================================================
    var x4 := ["hello", "world" , "monday", "tuesday", "wednesday", "thursday", "friday"]; 
    print x4 \n; 
    
    i := 0;
    while (i < length x4) {
        print [x4:i] \t; 
        i := i + 1; 
    }
    print \n; 
    
    print [x4:(length x4-1)] == "friday" \n;
    print [x4:(length x4-1)] == [x4:(arrayLen-1)] \n; 
    print \n; 
       
    
    
    # TEST CHAR ARRAY =========================================================
    var x5 := ['a', 'b' , %100, %122, 't', %176, '<']; 
    print x5 \n; 
    
    i := 0;
    while (i < length x5) {
        print [x5:i] \t; 
        i := i + 1; 
    }
    print \n; 
    
    print [x5:3] == %122 \n; 
    print [x5:3] == 'R' \n;
    print [x5:1] == %122 \n; 
    print \n;   
    
    
    # TEST BOOL ARRAY =========================================================
    var x6 := [true, false , true, true, true, false, true]; 
    print x6 \n; 
    
    i := 0;
    while (i < length x6) {
        print [x6:i] \t; 
        i := i + 1; 
    }
    print \n;  
    
    print [x6:0] && true \n; 
    print [x6:1] && true \n; 
    
    print [x6:0] || false \n; 
    print [x6:1] || false \n;
    print \n; 
    
   
}