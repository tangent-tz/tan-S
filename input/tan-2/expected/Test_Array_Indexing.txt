start
[ 1, 2, 33, 4, 5, 6, 777777 ]
1	2	33	4	5	6	777777	
true
true
false

[ 1.340000, 1.000000, 33.200000, 4.400000, 5.567000, -6.545600, -0.200000 ]
1.340000	1.000000	33.200000	4.400000	5.567000	-6.545600	-0.200000	
true
true

[ hello, world, monday, tuesday, wednesday, thursday, friday ]
hello	world	monday	tuesday	wednesday	thursday	friday	
false
true

[ a, b, @, R, t, ~, < ]
a	b	@	R	t	~	<	
true
true
false

[ true, false, true, true, true, false, true ]
true	false	true	true	true	false	true	
true
false
true
false

[ 1, 2 ]
[ 9, 10 ]
[ 1.100000, 2.200000 ]
[ 9.900000, 10.100000 ]
[ hello, hi ]
[ 1989, invisible string ]
[ a, b ]
[ ~, ! ]
[ true, true ]
[ false, true ]
